controller/controlplane/controller.go

```diff
@@ -130,6 +134,10 @@ func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Resu
 		return ctrl.Result{}, client.IgnoreNotFound(err)
 	}
 
+	if err := r.cleanupOldManagedResources(ctx, cp); err != nil {
+		return ctrl.Result{}, fmt.Errorf("failed to cleanup old managed resources: %w", err)
+	}
+
```

Comment:
Should we run this logic only once?

Reply:
We could add an annotation after a successful cleanup but that would add complexity and the client calls are using cache anyway. Do you think that would be worth the hassle?

Reply:
Why not tracking the state in the reconciler? A simple bool should do the trick and avoid to run the cleanup at every reconcile.